core/src/main/java/org/wildfly/plugin/core/GalleonUtils.java

```diff
@@ -140,10 +166,21 @@ public static ProvisioningConfig buildConfig(ProvisioningManager pm,
             final FeaturePackConfig.Builder fpConfig = fp.isTransitive() ? FeaturePackConfig.transitiveBuilder(fpl)
                     : FeaturePackConfig.builder(fpl);
             if (fp.isInheritConfigs() == null) {
-                fpConfig.setInheritConfigs(false);
+                if (hasLayers) {
+                    fpConfig.setInheritConfigs(false);
+                } else {
+                    if (fpWithDefaults) {
+                        if (isLastFP) {
+                            fpConfig.setInheritConfigs(true);
+                        } else {
+                            fpConfig.setInheritConfigs(false);
+                        }
+                    }
+                }
             } else {
                 fpConfig.setInheritConfigs(fp.isInheritConfigs());
             }
+            // We are optimistic, all default configs should be defined with Layers so we don't need to inherit default packages.
```

Comment:
Yes, that is correct. The default configs defined fully with layers (that is the goal), don't require to inherit packages. When inheriting packages, all JBoss modules get installed even the not needed ones.

Reply:
Just for me to understand. There is some mapping between config-->packages?

If layer has defined config snippet, that automatically determine which packages it brings in?

That is why one can use config when talking about packages, because config identify packages unambiguously?